File: go/safadometro.go
package safadometro

func WesleySafadao(day, month, year int) (float64, float64) {
  sum := 0
  y := float64(year)
  for month > 0 {
    sum += month
    month--
  }
  safadeza := float64(sum) + (y / 100.0) * float64(50 - day)
  anjo := 100.0 - safadeza
  return safadeza, anjo
}

File: nim/safadometro.nim
proc somatorio(n: int): int =
  result = 0
  for i in 1..n:
    result += i

proc wesley_safadao(dia, mes, ano: int) =
  var safadeza = float(somatorio mes) + (ano / 100) * float(50 - dia)
  var anjo = 100 - safadeza

  echo "Você é ", anjo, "% anjo e ", safadeza, "% safado."

when isMainModule:
  wesley_safadao 6, 9, 88
